jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -71,6 +71,9 @@
 	static final boolean enableCompression = com.ibm.oti.vm.VM.J9_STRING_COMPRESSION_ENABLED;
 	
 	/*[IF Sidecar19-SE]*/
+	/*[IF Java18.3]*/
+	private final byte coder = coder();
```

Comment:
The timing of this initialization is problematic: it occurs before the body of any constructor and so it will observe `count == 0` yielding the wrong answer for uncompressed strings.

Reply:
Moving `coder` initialization to inside of `j.l.String` constructors.